pkgs/by-name/pe/petsc/package.nix

```
@@ -62,6 +62,7 @@ stdenv.mkDerivation rec {
     "--CC=mpicc"
     "--with-cxx=mpicxx"
     "--with-fc=mpif90"
+    "--with-mpiexec=${mpi}/bin/mpiexec"
```

This will not work with cross as just ${mpi} refers to `pkgsHostTarget` version

Try this

```suggestion
    "--with-mpiexec=mpiexec"
```

If that doesn't work then `buildPackages.mpi`